tests.lua

```diff
@@ -66,6 +66,57 @@ do
     assert(tab.nested.c == nil)
 end
 
+do
+    local t1 = { x = "y", "test", [false] = { 1, 2, 3, a = "b" } }
+    local opts = {
+        stable = true,
+        filter = function(t, k, v) return not tonumber(k) or tonumber(k) < 100 end
```

Comment:
Will this filter actually reject anything in `t1`?

Reply:
Ah I get it, you're rejecting the keys you add to `t2`.

Reply:
Your filter didn't catch the bug from the first iter, though - if you use a nonserializable value instead of `i` for the filtered values, it should catch it.